ui: Update Locality hierarchy on Node List

koorosh opened the pull request:
Fixes: https://github.com/cockroachdb/cockroach/issues/44919
Depends on: https://github.com/cockroachdb/cockroach/pull/45007

Before, Node list with geo partitioned nodes was
grouped by 'region' key what was incorrect due to
locality keys might have arbitrary names and it
doesn't represent the lowest locality tier.

Now, nodes are grouped by lowest level tier and
also display all tiers in tooltip.

Preview:
<img width="722" alt="Screenshot 2020-02-12 at 23 13 03" src="https://user-images.githubusercontent.com/3106437/74383887-c163e900-4df8-11ea-9822-134edbfa520c.png">

koorosh committed:
ui: Update locality hierarchy on Node List

Before, Node list with geo partitioned nodes was
grouped by 'region' key what was incorrect due to
locality keys might have arbitrary names and it
doesn't represent the lowest locality tier.

Now, nodes are grouped by lowest level tier and
also display all tiers in tooltip.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45041)
<!-- Reviewable:end -->

dhartunian commented on pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx:
```diff
@@ -133,18 +137,30 @@ export class NodeList extends React.Component<LiveNodeListProps> {
-              <Text>{record.region}</Text>
-            </React.Fragment>);
+            <Text>
+              <Tooltip
```
Would be nice to extract the tooltip to a pure presentation component

koorosh replied:
Sorry if I wrong, but Tooltip component is already defined as pure component.
In this case we provide only props (`title` and `children`) as inputs. Both of these props has
to be defined outside of tooltip component.
To make it more readable, I would extract entire render component.

koorosh replied:
@dhartunian , please check if new changes somehow fit to yours proposed changes 🙄

dhartunian commented on pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx:
```diff
@@ -375,9 +388,18 @@ export const liveNodesTableDataSelector = createSelector(
         return {
           key: `${regionKey}`,
-          region: regionKey,
+          region: location?.value,
```
why do we call this `location`? Why not `lastTier`?

dhartunian commented on pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx:
```diff
@@ -133,18 +137,30 @@ export class NodeList extends React.Component<LiveNodeListProps> {
-                <Text>{record.region}</Text>
-              </Link>
-            </React.Fragment>);
+            <Link className="nodes-table__link" to={`/node/${record.nodeId}`}>
```
Can this `<Link>` tag be extracted to match the component on line 303 below?

koorosh committed:
ui: Extract NodeName and NodeLocality columns as external components

- NodeName and NodeLocality columns are refactored to external
components. It allows to reuse them in NodeList and Decommissioned
Node List components.
- Fixed naming variables.

Release note: None

dhartunian commented:
bors r+
